addons/controllers/clusterbootstrap_controller.go

```diff
@@ -902,6 +927,41 @@ func (r *ClusterBootstrapReconciler) createOrPatchPackageInstallSecretOnRemote(c
 		for k, v := range patchedSecret.Data {
 			remoteSecret.Data[k] = v
 		}
+		// Add TKR NodeSelector info if it's a TKGS cluster
+		infraRef, err := util.GetInfraProvider(cluster)
+		if err != nil {
+			return err
+		}
+		if infraRef == tkgconstants.InfrastructureProviderVSphere {
+			ok, err := util.IsTKGSCluster(r.context, r.Client, cluster)
+			if err != nil {
+				return err
+			}
+			if ok {
+				upgradeDataValues := addontypes.UpgradeDataValues{
+					NodeSelector: addontypes.NodeSelector{
+						TanzuKubernetesRelease: cluster.Labels[constants.TKRLabelClassyClusters],
+					},
+					Deployment: addontypes.DeploymentUpdateInfo{
+						UpdateStrategy: constants.TKGSDeploymentUpdateStrategy,
+						RollingUpdate: &addontypes.RollingUpdateInfo{
+							MaxSurge:       constants.TKGSDeploymentUpdateMaxSurge,
+							MaxUnavailable: constants.TKGSDeploymentUpdateMaxUnavailable,
+						},
+					},
+					Daemonset: addontypes.DaemonsetUpdateInfo{
+						UpdateStrategy: constants.TKGSDaemonsetUpdateStrategy,
+					},
+				}
+				TKRDataValueYamlBytes, err := yaml.Marshal(upgradeDataValues)
+				if err != nil {
+					return err
+				}
+				// TODO: Make sure either YTT schema don't care about unspecified field, or all packages have `nodeSelector` field by default in schema.yaml
```

Comment:
I guess we need to check if packages have a schema and if they contain the data values being added here.

Reply:
Since we are adding an additional feature flag for it, removing this comment